schema/dmt/compile.go

```diff
@@ -134,9 +161,42 @@ func spawnType(ts *schema.TypeSystem, name schema.TypeName, defn TypeDefn) (sche
 				todoFromImplicitlyFalseBool(field.Nullable),
 			))
 		}
+		var repr schema.StructRepresentation
+		switch {
+		case typ.Representation.StructRepresentation_Map != nil:
+			rp := typ.Representation.StructRepresentation_Map
+			if rp.Fields == nil {
+				repr = schema.SpawnStructRepresentationMap2(nil, nil)
+				break
+			}
+			renames := make(map[string]string, len(rp.Fields.Keys))
+			implicits := make(map[string]schema.ImplicitValue, len(rp.Fields.Keys))
+			for _, name := range rp.Fields.Keys {
+				details := rp.Fields.Values[name]
+				if details.Rename != nil {
+					renames[name] = *details.Rename
+				}
+				if imp := details.Implicit; imp != nil {
+					var sumVal schema.ImplicitValue
+					switch {
+					case imp.Bool != nil:
+						sumVal = schema.ImplicitValue_Bool(*imp.Bool)
+					case imp.String != nil:
+						sumVal = schema.ImplicitValue_String(*imp.String)
+					default:
+						panic("TODO: implicit value kind")
+					}
+					implicits[name] = sumVal
+				}
+
+			}
+			repr = schema.SpawnStructRepresentationMap2(renames, implicits)
+		default:
```

Comment:
fwiw, if you do feel like continuing to strike hereabouts while the iron is hot, I think the tuple representation strategy is a relatively highly used one.  It's used in the filecoin protocol, heavily, certainly, at least.

This error is fine and explicit for now though.  Things are still moving in the right direction with this diff.

Reply:
Right, there are quite a lot of repr TODOs left, but we can incrementally work on those.